tests/safety/test_ford.py

```diff
@@ -234,28 +238,84 @@ def test_steer_allowed(self):
                                   curvature=curvature):
                   self.assertEqual(should_tx, self._tx(self._tja_command_msg(steer_control_enabled, path_offset, path_angle, curvature, curvature_rate)))
 
-  def test_steer_meas_delta(self):
-    """This safety model enforces a maximum distance from measured and commanded curvature, only above a certain speed"""
-    self.safety.set_controls_allowed(1)
-
-    for steer_control_enabled in (True, False):
-      for speed in np.linspace(0, 50, 11):
-        for initial_curvature in np.linspace(-self.MAX_CURVATURE, self.MAX_CURVATURE, 21):
-          self._reset_curvature_measurement(initial_curvature, speed)
-
-          limit_command = speed > self.CURVATURE_DELTA_LIMIT_SPEED
-          for new_curvature in np.linspace(-self.MAX_CURVATURE, self.MAX_CURVATURE, 41):
-            too_far_away = round_curvature_can(abs(new_curvature - initial_curvature)) > self.MAX_CURVATURE_DELTA
-
-            if steer_control_enabled:
-              should_tx = not limit_command or not too_far_away
-            else:
-              # enforce angle error limit is disabled when steer request bit is 0
-              should_tx = new_curvature == 0
-
-            with self.subTest(steer_control_enabled=steer_control_enabled, speed=speed,
-                              initial_curvature=initial_curvature, new_curvature=new_curvature):
-              self.assertEqual(should_tx, self._tx(self._tja_command_msg(steer_control_enabled, 0, 0, new_curvature, 0)))
+  def test_curvature_rate_limit_up(self):
+    """
+    When the curvature error is exceeded, commanded curvature must start moving towards meas respecting rate limits.
+    Since panda allows higher rate limits to avoid false positives, we need to allow a lower rate to move towards meas.
+    """
+    self.safety.set_controls_allowed(True)
+    small_curvature = 2 / self.DEG_TO_CAN  # significant small amount of curvature to cross boundary
+    curvature_meas = self.MAX_CURVATURE_DELTA + 1e-3
+
+    for speed in np.arange(0, 41, 5):
+      limit_command = speed > self.CURVATURE_DELTA_LIMIT_SPEED
+      max_delta_up = np.interp(speed, self.ANGLE_DELTA_BP, self.ANGLE_DELTA_V)
+      max_delta_up_lower = np.interp(speed + 1, self.ANGLE_DELTA_BP, self.ANGLE_DELTA_V)
+
+      for sign in (-1, 1):
+        self._reset_curvature_measurement(sign * curvature_meas, speed)
+
+        # stay the same
+        # TODO: small_curvature instead of zero since panda allows down limits at 0 to mask a Tesla bug
```

Comment:
let's fix this bug (sounds like we're just not doing the signal offset?). it should not be leaking into the tests of another safety mode

Reply:
not a bug, forgot that allowing down limits at 0 prevents some false positive blocking as small floats will be floored to 0 over CAN, so we allow the higher down limits in case we want to go across zero with the down limits